src/Service/RequestService.php

```diff
@@ -538,34 +559,66 @@ public function proxyHandler(array $data, array $configuration, ?Source $proxy =
+            // Make sure we do not send this gateway specific query param to the proxy / Source.
+            unset($this->data['query']['extend']);
+        }
+
+        // Make sure we set object to null in the session, for detecting the correct AuditTrails to create. Also used for DateRead to work correctly!
+        $this->session->set('object', null);
+
         if (isset($data['path']['{route}']) === true) {
             $this->data['path'] = '/'.$data['path']['{route}'];
         } else {
             $this->data['path'] = '';
         }
 
+        // Don't pass gateway authorization to the source
         unset($this->data['headers']['authorization']);
+
+        $url = \Safe\parse_url($proxy->getLocation());
+
         // Make a guzzle call to the source based on the incoming call.
         try {
-            $result = $this->callService->call(
-                $proxy,
-                $this->data['path'],
-                $this->data['method'],
-                [
-                    'query'   => $this->data['query'],
-                    'headers' => $this->data['headers'],
-                    'body'    => $this->data['crude_body'],
-                ]
-            );
+            // Check if we are dealing with http, https or something else like a ftp (fileSystem)
+            if ($url['scheme'] === 'http' || $url['scheme'] === 'https') {
+                $result = $this->callService->call(
+                    $proxy,
+                    $this->data['path'],
+                    $this->data['method'],
+                    [
+                        'query'   => $this->data['query'],
+                        'headers' => $this->data['headers'],
+                        'body'    => $this->data['crude_body'],
+                    ]
+                );
+            } else {
+                $result = $this->fileSystemService->call($proxy, $this->data['path']);
+                $result = new \GuzzleHttp\Psr7\Response(200, [], $this->serializer->serialize($result, 'json'));
+            }
+
+            $resultContent = $this->unserializeData($result->getBody()->getContents(), $result->getHeaders()['content-type'][0]);
+
+            // Handle _self metadata, includes adding dateRead
+            if (isset($extend) === true) {
+                $this->data['query']['extend'] = $extend;
+            }
+
+            $this->handleMetadataSelf($resultContent, $proxy);
 
             // Let create a response from the guzzle call.
             $response = new Response(
-                $result->getBody()->getContents(),
+                $this->serializeData($resultContent, $result->getHeaders()['content-type'][0]),
                 $result->getStatusCode(),
                 $result->getHeaders()
             );
         } catch (Exception $exception) {
-            $statusCode = ($exception->getCode() ?? 500);
+            $statusCode = 500;
+            if (array_key_exists($exception->getCode(), Response::$statusTexts)) {
```

Comment:
Interesting way of determining if there is a status code

Reply:
Is there a better way of doing this?
I liked this one 😛

Reply:
Not sure, I'm not saying it is wrong, just interesting 😛